diagram-viewer/src/main/java/com/powsybl/diagram/viewer/MainViewController.java

```diff
@@ -187,12 +239,39 @@ protected Network call() {
                     };
                 }
             };
-            handleLoadingResult(file, networkService);
+            handleLoadingFileResult(file, networkService);
+            networkService.start();
+        }
+    }
+
+    public void loadFactory(Class<?> clazz) {
+        if (clazz != null) {
+            Service<Network> networkService = new Service<>() {
+                @Override
+                protected Task<Network> createTask() {
+                    return new Task<>() {
+                        @Override
+                        protected Network call() throws InstantiationException, IllegalAccessException, IllegalArgumentException, java.lang.reflect.InvocationTargetException, NoSuchMethodException, SecurityException {
+                            // Get constructor
+                            Constructor<?> constructor = clazz.getDeclaredConstructor();
+                            constructor.trySetAccessible();
+                            // Make new instance
+                            Object instance = constructor.newInstance();
+                            // Get create() method using refection
+                            Method createMethod = instance.getClass().getDeclaredMethod("create");
+                            createMethod.trySetAccessible();
+                            // Get Network instance
+                            return (Network) createMethod.invoke(instance);
+                        }
```

Comment:
You're using heavy powerful material here, it made sense if you were using the reflections, but now that you're writing the list class by hand you'd better make it even simpler and provide directly the list of `Supplier<Network>`, that is 
```
List.of(
  BatteryNetworkFactory::create, 
  BusbarSectionExt::create,
...
```

Reply:
Thanks for the tips !